scene/main/window.cpp

```diff
@@ -254,7 +254,10 @@ void Window::_make_window() {
 #endif
 	DisplayServer::get_singleton()->window_set_title(tr_title, window_id);
 	DisplayServer::get_singleton()->window_attach_instance_id(get_instance_id(), window_id);
-
+	// Inherit the window icon from the main window if the new sub_window is configured with a title bar
+	if (!get_flag(FLAG_BORDERLESS)) {
+		DisplayServer::get_singleton()->inherit_icon(window_id);
+	}
```

Comment:
I'm not sure exposing `inherit_icon` is useful, just doing it for the all new windows (probably in the `DispalyServerWindows`) should be enough, it's a reasonable default behavior.

And `set_native_icon`/`set_icon` probably should set icon for all windows (or have an optional argument to set individual window icon).

Reply:
I was originally going to modify the set_icon and set_native_icon methods to take a window_id parameter, but thought that reading the icon handles form the existing parent window (main window) may be quicker/easier than re-parsing the icon input file (either the image or native files).

Happy to go back to that thinking if it's the preferred solution.

Reply:
> I'm not sure exposing `inherit_icon` is useful, just doing it for the all new windows (probably in the `DispalyServerWindows`) should be enough, it's a reasonable default behavior.

The `_create_sub_window` method could be modified to include the read-icon-from-main / write-icon-to-sub-window code in place of the new inherit_icon code. That certainly wraps it up a little neater and doesn't add another method to the public namespace.